### ISSUE TITLE
Workflow not exiting even both pod and workflow completed

### ISSUE THREAD

**Issue by jackiemazy** [argoproj/argo#2884]
Checklist:

 * [x] I've included the version.
 * [x] I've included reproduction steps.
 * [x] I've included the workflow YAML.
 * [x] I've included the logs.
 
**What happened**:
While submitting a simple argo workflow with only one step, the pod gets created and completed successfully, so is the workflow, but the workflow doesn't exit. it hangs at the following stage:
```
$argo submit ./nrdee-indexer-workflow.yaml --watch --serviceaccount=argo
Name:                nrdee-indexer-rzg6f
Namespace:           asias-etl
ServiceAccount:      argo
Status:              Running
Created:             Wed Apr 29 07:37:44 -0400 (7 seconds ago)
Started:             Wed Apr 29 07:37:44 -0400 (7 seconds ago)
Duration:            7 seconds

STEP                    TEMPLATE             PODNAME                         DURATION  MESSAGE
 ● nrdee-indexer-rzg6f  run-nrdee-indexer
 └---● nrdee-indexer    run-indexer-process  nrdee-indexer-rzg6f-1673954213  7s
```

**What you expected to happen**:
The workflow should complete and exit.

**How to reproduce it (as minimally and precisely as possible)**:
Run argo submit with the following yaml workflow file:
```
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: nrdee-indexer-
spec:
  entrypoint: run-nrdee-indexer
  volumes:
  - name: asias-etl-nrdee-solr-volume-pv
    persistentVolumeClaim:
      claimName: asias-etl-nrdee-solr-volume

  templates:
  - name: run-nrdee-indexer
    steps:
    - - name: nrdee-indexer
        template: run-indexer-process

  - name: run-indexer-process
    container:
      image: docker-registry-default.okd.mitre.org/asias-etl/nrdee-indexer:latest
      name: nrdee-indexer
      volumeMounts:
      - name: asias-etl-nrdee-solr-volume-pv
        mountPath: /var/lib/nrdee/solr
      resources:
        limits:
          cpu: 1
          memory: 5Gi
        requests:
          cpu: 1
          memory: 5Gi
```
**Anything else we need to know?**:
There are hundreds of thousands lines of logs in the pod main container, not sure if this could be an issue. Please let me know if you need more info here... Thanks for looking into this.

**Environment**:

- Argo version:
```
$ argo version
argo: vv2.7.4+50b209c.dirty
  BuildDate: 2020-04-16T16:27:43Z
  GitCommit: 50b209ca14c056fb470ebb8329e255304dd5be90
  GitTreeState: dirty
  GitTag: v2.7.4
  GoVersion: go1.13
  Compiler: gc
  Platform: linux/amd64
```
- Kubernetes version :
```
$ oc version
oc v3.11.0+0cbc58b
kubernetes v1.11.0+d4cacc0
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://okd-load.mitre.org:8443
openshift v3.11.0+1c3e643-87
kubernetes v1.11.0+d4cacc0
```

**Other debugging information (if applicable)**:

- workflow result:

```
$ argo --loglevel DEBUG get nrdee-indexer-rzg6f
Name:                nrdee-indexer-rzg6f
Namespace:           asias-etl
ServiceAccount:      argo
Status:              Succeeded
Created:             Wed Apr 29 07:37:44 -0400 (1 hour ago)
Started:             Wed Apr 29 07:37:44 -0400 (1 hour ago)
Finished:            Wed Apr 29 07:58:21 -0400 (1 hour ago)
Duration:            20 minutes 37 seconds

STEP                    TEMPLATE             PODNAME                         DURATION  MESSAGE
 ✔ nrdee-indexer-rzg6f  run-nrdee-indexer
 └---✔ nrdee-indexer    run-indexer-process  nrdee-indexer-rzg6f-1673954213  20m
```

- executor logs:

```
$oc logs nrdee-indexer-rzg6f-1673954213 -c init
Error from server (BadRequest): container init is not valid for pod nrdee-indexer-rzg6f-1673954213
(I guess it's because the pod is already completed?)

$ oc logs nrdee-indexer-rzg6f-1673954213 -c wait
time="2020-04-29T11:37:51Z" level=info msg="Starting Workflow Executor" version=vv2.7.4+50b209c.dirty
time="2020-04-29T11:37:51Z" level=info msg="Creating a K8sAPI executor"
time="2020-04-29T11:37:51Z" level=info msg="Executor (version: vv2.7.4+50b209c.dirty, build_date: 2020-04-16T16:37:57Z) initialized (pod: asias-etl/nrdee-indexer-rzg6f-1673954213) with template:\n{\"name\":\"run-indexer-process\",\"arguments\":{},\"inputs\":{},\"outputs\":{},\"metadata\":{},\"container\":{\"name\":\"nrdee-indexer\",\"image\":\"docker-registry-default.okd.mitre.org/asias-etl/nrdee-indexer:latest\",\"resources\":{\"limits\":{\"cpu\":\"1\",\"memory\":\"5Gi\"},\"requests\":{\"cpu\":\"1\",\"memory\":\"5Gi\"}},\"volumeMounts\":[{\"name\":\"secret\",\"mountPath\":\"/etc/confd/.container/secrets\"},{\"name\":\"asias-etl-nrdee-solr-volume-pv\",\"mountPath\":\"/var/lib/nrdee/solr\"}]}}"
time="2020-04-29T11:37:51Z" level=info msg="Waiting on main container"
time="2020-04-29T11:37:51Z" level=info msg="main container started with container ID: b94bda272c0bac6e6549befa05efd703c885814957e19f49994a06c52275370c"
time="2020-04-29T11:37:51Z" level=info msg="Starting annotations monitor"
time="2020-04-29T11:37:51Z" level=info msg="Waiting for container b94bda272c0bac6e6549befa05efd703c885814957e19f49994a06c52275370c to complete"
time="2020-04-29T11:37:51Z" level=info msg="Starting to wait completion of containerID b94bda272c0bac6e6549befa05efd703c885814957e19f49994a06c52275370c ..."
time="2020-04-29T11:37:51Z" level=info msg="Starting deadline monitor"
time="2020-04-29T11:42:51Z" level=info msg="Alloc=5501 TotalAlloc=26613 Sys=71104 NumGC=9 Goroutines=10"
time="2020-04-29T11:47:51Z" level=info msg="Alloc=5165 TotalAlloc=41988 Sys=71104 NumGC=14 Goroutines=10"
time="2020-04-29T11:52:51Z" level=info msg="Alloc=4729 TotalAlloc=57392 Sys=71104 NumGC=19 Goroutines=10"
time="2020-04-29T11:57:51Z" level=info msg="Alloc=4085 TotalAlloc=72713 Sys=71360 NumGC=24 Goroutines=10"
time="2020-04-29T11:58:20Z" level=info msg="ContainerID \"b94bda272c0bac6e6549befa05efd703c885814957e19f49994a06c52275370c\" is terminated: &ContainerStatus{Name:main,State:ContainerState{Waiting:nil,Running:nil,Terminated:&ContainerStateTerminated{ExitCode:0,Signal:0,Reason:Completed,Message:,StartedAt:2020-04-29 11:37:51 +0000 UTC,FinishedAt:2020-04-29 11:58:19 +0000 UTC,ContainerID:docker://b94bda272c0bac6e6549befa05efd703c885814957e19f49994a06c52275370c,},},LastTerminationState:ContainerState{Waiting:nil,Running:nil,Terminated:nil,},Ready:false,RestartCount:0,Image:docker-registry-default.okd.mitre.org/asias-etl/nrdee-indexer:latest,ImageID:docker-pullable://docker-registry-default.okd.mitre.org/asias-etl/nrdee-indexer@sha256:f15ebeed2a82850ccbaf6c2dd1e2948611fb19cf5ef5e1ed616aa473aff27bd8,ContainerID:docker://b94bda272c0bac6e6549befa05efd703c885814957e19f49994a06c52275370c,Started:nil,}"
time="2020-04-29T11:58:20Z" level=info msg="Main container completed"
time="2020-04-29T11:58:20Z" level=info msg="No output parameters"
time="2020-04-29T11:58:20Z" level=info msg="No output artifacts"
time="2020-04-29T11:58:20Z" level=info msg="No Script output reference in workflow. Capturing script output ignored"
time="2020-04-29T11:58:20Z" level=info msg="Killing sidecars"
time="2020-04-29T11:58:20Z" level=info msg="Annotations monitor stopped"
time="2020-04-29T11:58:20Z" level=info msg="Alloc=5748 TotalAlloc=74377 Sys=71360 NumGC=24 Goroutines=9"
```

- workflow-controller logs:

```
$ oc logs workflow-controller-1-72fzl --since=40m
time="2020-04-29T11:54:17Z" level=info msg="Processing workflow" namespace=asias-etl workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:54:17Z" level=info msg="Workflow step group node &NodeStatus{ID:nrdee-indexer-rzg6f-2973924931,Name:nrdee-indexer-rzg6f[0],DisplayName:[0],Type:StepGroup,TemplateName:run-nrdee-indexer,TemplateRef:nil,Phase:Running,BoundaryID:nrdee-indexer-rzg6f,Message:,StartedAt:2020-04-29 11:37:44 +0000 UTC,FinishedAt:0001-01-01 00:00:00 +0000 UTC,PodIP:,Daemoned:nil,Inputs:nil,Outputs:nil,Children:[nrdee-indexer-rzg6f-1673954213],OutboundNodes:[],StoredTemplateID:,WorkflowTemplateName:,TemplateScope:,ResourcesDuration:ResourcesDuration{},} not yet completed" namespace=asias-etl workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:57:10Z" level=info msg="Alloc=4868 TotalAlloc=1150055 Sys=72640 NumGC=4200 Goroutines=101"
W0429 11:57:14.383826       1 reflector.go:302] pkg/mod/k8s.io/client-go@v0.0.0-20191225075139-73fd2ddc9180/tools/cache/reflector.go:98: watch of *unstructured.Unstructured ended with: The resourceVersion for the provided watch is too old.
W0429 11:58:10.822946       1 reflector.go:302] pkg/mod/k8s.io/client-go@v0.0.0-20191225075139-73fd2ddc9180/tools/cache/reflector.go:98: watch of *v1.ConfigMap ended with: The resourceVersion for the provided watch is too old.
time="2020-04-29T11:58:20Z" level=info msg="Processing workflow" namespace=asias-etl workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:58:20Z" level=info msg="Workflow step group node &NodeStatus{ID:nrdee-indexer-rzg6f-2973924931,Name:nrdee-indexer-rzg6f[0],DisplayName:[0],Type:StepGroup,TemplateName:run-nrdee-indexer,TemplateRef:nil,Phase:Running,BoundaryID:nrdee-indexer-rzg6f,Message:,StartedAt:2020-04-29 11:37:44 +0000 UTC,FinishedAt:0001-01-01 00:00:00 +0000 UTC,PodIP:,Daemoned:nil,Inputs:nil,Outputs:nil,Children:[nrdee-indexer-rzg6f-1673954213],OutboundNodes:[],StoredTemplateID:,WorkflowTemplateName:,TemplateScope:,ResourcesDuration:ResourcesDuration{},} not yet completed" namespace=asias-etl workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:58:21Z" level=info msg="Processing workflow" namespace=asias-etl workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:58:21Z" level=info msg="Updating node &NodeStatus{ID:nrdee-indexer-rzg6f-1673954213,Name:nrdee-indexer-rzg6f[0].nrdee-indexer,DisplayName:nrdee-indexer,Type:Pod,TemplateName:run-indexer-process,TemplateRef:nil,Phase:Running,BoundaryID:nrdee-indexer-rzg6f,Message:,StartedAt:2020-04-29 11:37:44 +0000 UTC,FinishedAt:0001-01-01 00:00:00 +0000 UTC,PodIP:,Daemoned:nil,Inputs:nil,Outputs:nil,Children:[],OutboundNodes:[],StoredTemplateID:,WorkflowTemplateName:,TemplateScope:,ResourcesDuration:ResourcesDuration{},} status Running -> Succeeded"
time="2020-04-29T11:58:21Z" level=info msg="Step group node &NodeStatus{ID:nrdee-indexer-rzg6f-2973924931,Name:nrdee-indexer-rzg6f[0],DisplayName:[0],Type:StepGroup,TemplateName:run-nrdee-indexer,TemplateRef:nil,Phase:Running,BoundaryID:nrdee-indexer-rzg6f,Message:,StartedAt:2020-04-29 11:37:44 +0000 UTC,FinishedAt:0001-01-01 00:00:00 +0000 UTC,PodIP:,Daemoned:nil,Inputs:nil,Outputs:nil,Children:[nrdee-indexer-rzg6f-1673954213],OutboundNodes:[],StoredTemplateID:,WorkflowTemplateName:,TemplateScope:,ResourcesDuration:ResourcesDuration{},} successful" namespace=asias-etl workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:58:21Z" level=info msg="node &NodeStatus{ID:nrdee-indexer-rzg6f-2973924931,Name:nrdee-indexer-rzg6f[0],DisplayName:[0],Type:StepGroup,TemplateName:run-nrdee-indexer,TemplateRef:nil,Phase:Running,BoundaryID:nrdee-indexer-rzg6f,Message:,StartedAt:2020-04-29 11:37:44 +0000 UTC,FinishedAt:0001-01-01 00:00:00 +0000 UTC,PodIP:,Daemoned:nil,Inputs:nil,Outputs:nil,Children:[nrdee-indexer-rzg6f-1673954213],OutboundNodes:[],StoredTemplateID:,WorkflowTemplateName:,TemplateScope:,ResourcesDuration:ResourcesDuration{},} phase Running -> Succeeded" namespace=asias-etl workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:58:21Z" level=info msg="node &NodeStatus{ID:nrdee-indexer-rzg6f-2973924931,Name:nrdee-indexer-rzg6f[0],DisplayName:[0],Type:StepGroup,TemplateName:run-nrdee-indexer,TemplateRef:nil,Phase:Succeeded,BoundaryID:nrdee-indexer-rzg6f,Message:,StartedAt:2020-04-29 11:37:44 +0000 UTC,FinishedAt:2020-04-29 11:58:21.295230666 +0000 UTC,PodIP:,Daemoned:nil,Inputs:nil,Outputs:nil,Children:[nrdee-indexer-rzg6f-1673954213],OutboundNodes:[],StoredTemplateID:,WorkflowTemplateName:,TemplateScope:,ResourcesDuration:ResourcesDuration{},} finished: 2020-04-29 11:58:21.295230666 +0000 UTC" namespace=asias-etl workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:58:21Z" level=info msg="Outbound nodes of nrdee-indexer-rzg6f-1673954213 is [nrdee-indexer-rzg6f-1673954213]" namespace=asias-etl workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:58:21Z" level=info msg="Outbound nodes of nrdee-indexer-rzg6f is [nrdee-indexer-rzg6f-1673954213]" namespace=asias-etl workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:58:21Z" level=info msg="node &NodeStatus{ID:nrdee-indexer-rzg6f,Name:nrdee-indexer-rzg6f,DisplayName:nrdee-indexer-rzg6f,Type:Steps,TemplateName:run-nrdee-indexer,TemplateRef:nil,Phase:Running,BoundaryID:,Message:,StartedAt:2020-04-29 11:37:44 +0000 UTC,FinishedAt:0001-01-01 00:00:00 +0000 UTC,PodIP:,Daemoned:nil,Inputs:nil,Outputs:nil,Children:[nrdee-indexer-rzg6f-2973924931],OutboundNodes:[nrdee-indexer-rzg6f-1673954213],StoredTemplateID:,WorkflowTemplateName:,TemplateScope:,ResourcesDuration:ResourcesDuration{},} phase Running -> Succeeded" namespace=asias-etl workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:58:21Z" level=info msg="node &NodeStatus{ID:nrdee-indexer-rzg6f,Name:nrdee-indexer-rzg6f,DisplayName:nrdee-indexer-rzg6f,Type:Steps,TemplateName:run-nrdee-indexer,TemplateRef:nil,Phase:Succeeded,BoundaryID:,Message:,StartedAt:2020-04-29 11:37:44 +0000 UTC,FinishedAt:2020-04-29 11:58:21.295327755 +0000 UTC,PodIP:,Daemoned:nil,Inputs:nil,Outputs:nil,Children:[nrdee-indexer-rzg6f-2973924931],OutboundNodes:[nrdee-indexer-rzg6f-1673954213],StoredTemplateID:,WorkflowTemplateName:,TemplateScope:,ResourcesDuration:ResourcesDuration{},} finished: 2020-04-29 11:58:21.295327755 +0000 UTC" namespace=asias-etl workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:58:21Z" level=info msg="Checking daemoned children of nrdee-indexer-rzg6f" namespace=asias-etl workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:58:21Z" level=info msg="Updated phase Running -> Succeeded" namespace=asias-etl workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:58:21Z" level=info msg="Marking workflow completed" namespace=asias-etl workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:58:21Z" level=info msg="Checking daemoned children of " namespace=asias-etl workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:58:21Z" level=info msg="Workflow update successful" namespace=asias-etl phase=Succeeded resourceVersion=180809477 workflow=nrdee-indexer-rzg6f
time="2020-04-29T11:58:22Z" level=info msg="Labeled pod asias-etl/nrdee-indexer-rzg6f-1673954213 completed"
W0429 12:00:08.147863       1 reflector.go:302] pkg/mod/k8s.io/client-go@v0.0.0-20191225075139-73fd2ddc9180/tools/cache/reflector.go:98: watch of *unstructured.Unstructured ended with: The resourceVersion for the provided watch is too old.
W0429 12:00:38.649456       1 reflector.go:302] pkg/mod/k8s.io/client-go@v0.0.0-20191225075139-73fd2ddc9180/tools/cache/reflector.go:98: watch of *v1alpha1.WorkflowTemplate ended with: The resourceVersion for the provided watch is too old.
time="2020-04-29T12:02:10Z" level=info msg="Alloc=4851 TotalAlloc=1151570 Sys=72640 NumGC=4203 Goroutines=101"
W0429 12:03:38.239053       1 reflector.go:302] pkg/mod/k8s.io/client-go@v0.0.0-20191225075139-73fd2ddc9180/tools/cache/reflector.go:98: watch of *v1alpha1.CronWorkflow ended with: The resourceVersion for the provided watch is too old.
time="2020-04-29T12:07:10Z" level=info msg="Alloc=4967 TotalAlloc=1151924 Sys=72640 NumGC=4205 Goroutines=100"
```

**Status of the workflow and pod**
As described below, both the pod and workflow are completed successfully, but the workflow didn't exit.

Pod:
```
$ oc describe pod nrdee-indexer-rzg6f-1673954213
Name:               nrdee-indexer-rzg6f-1673954213
Namespace:          asias-etl
Priority:           0
PriorityClassName:  <none>
Node:               okd-comp1.mitre.org/128.29.34.147
Start Time:         Wed, 29 Apr 2020 07:37:44 -0400
Labels:             workflows.argoproj.io/completed=true
                    workflows.argoproj.io/workflow=nrdee-indexer-rzg6f
Annotations:        openshift.io/scc=restricted
                    workflows.argoproj.io/node-name=nrdee-indexer-rzg6f[0].nrdee-indexer
                    workflows.argoproj.io/template={"name":"run-indexer-process","arguments":{},"inputs":{},"outputs":{},"metadata":{},"container":{"name":"nrdee-indexer","image":"docker-registry-default.okd.mitre.org/as...
Status:             Succeeded
IP:                 10.131.2.128
Controlled By:      Workflow/nrdee-indexer-rzg6f
Containers:
  wait:
    Container ID:  docker://12b796aeb10e6d1313b16270900936232e996afb67a8993f3e3c9e6d2573f414
    Image:         argoproj/argoexec:v2.7.4
    Image ID:      docker-pullable://docker.io/argoproj/argoexec@sha256:3cc7dbdba70e3d4a184d85483dacae6fdde645a02b3be37e155af1448a1e2a4b
    Port:          <none>
    Host Port:     <none>
    Command:
      argoexec
      wait
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Wed, 29 Apr 2020 07:37:50 -0400
      Finished:     Wed, 29 Apr 2020 07:58:20 -0400
    Ready:          False
    Restart Count:  0
    Limits:
      cpu:     500m
      memory:  300Mi
    Requests:
      cpu:     500m
      memory:  300Mi
    Environment:
      ARGO_POD_NAME:                    nrdee-indexer-rzg6f-1673954213 (v1:metadata.name)
      ARGO_CONTAINER_RUNTIME_EXECUTOR:  k8sapi
    Mounts:
      /argo/podmetadata from podmetadata (rw)
      /mainctrfs/etc/confd/.container/secrets from secret (rw)
      /mainctrfs/var/lib/nrdee/solr from asias-etl-nrdee-solr-volume-pv (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from argo-token-v2bhw (ro)
  main:
    Container ID:   docker://b94bda272c0bac6e6549befa05efd703c885814957e19f49994a06c52275370c
    Image:          docker-registry-default.okd.mitre.org/asias-etl/nrdee-indexer:latest
    Image ID:       docker-pullable://docker-registry-default.okd.mitre.org/asias-etl/nrdee-indexer@sha256:f15ebeed2a82850ccbaf6c2dd1e2948611fb19cf5ef5e1ed616aa473aff27bd8
    Port:           <none>
    Host Port:      <none>
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Wed, 29 Apr 2020 07:37:51 -0400
      Finished:     Wed, 29 Apr 2020 07:58:19 -0400
    Ready:          False
    Restart Count:  0
    Limits:
      cpu:     1
      memory:  5Gi
    Requests:
      cpu:        1
      memory:     5Gi
    Environment:  <none>
    Mounts:
      /etc/confd/.container/secrets from secret (rw)
      /var/lib/nrdee/solr from asias-etl-nrdee-solr-volume-pv (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from argo-token-v2bhw (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  podmetadata:
    Type:  DownwardAPI (a volume populated by information about the pod)
    Items:
      metadata.annotations -> annotations
  secret:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  app-confd-secrets-nrdee-indexer
    Optional:    false
  asias-etl-nrdee-solr-volume-pv:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  asias-etl-nrdee-solr-volume
    ReadOnly:   false
  argo-token-v2bhw:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  argo-token-v2bhw
    Optional:    false
QoS Class:       Guaranteed
Node-Selectors:  node-role.kubernetes.io/compute=true
Tolerations:     node.kubernetes.io/memory-pressure:NoSchedule
Events:
  Type    Reason     Age   From                          Message
  ----    ------     ----  ----                          -------
  Normal  Scheduled  28m   default-scheduler             Successfully assigned asias-etl/nrdee-indexer-rzg6f-1673954213 to okd-comp1.mitre.org
  Normal  Pulled     28m   kubelet, okd-comp1.mitre.org  Container image "argoproj/argoexec:v2.7.4" already present on machine
  Normal  Created    28m   kubelet, okd-comp1.mitre.org  Created container
  Normal  Started    28m   kubelet, okd-comp1.mitre.org  Started container
  Normal  Pulling    28m   kubelet, okd-comp1.mitre.org  pulling image "docker-registry-default.okd.mitre.org/asias-etl/nrdee-indexer:latest"
  Normal  Pulled     28m   kubelet, okd-comp1.mitre.org  Successfully pulled image "docker-registry-default.okd.mitre.org/asias-etl/nrdee-indexer:latest"
  Normal  Created    28m   kubelet, okd-comp1.mitre.org  Created container
  Normal  Started    28m   kubelet, okd-comp1.mitre.org  Started container
```
Workflow:
```
$ oc describe workflow nrdee-indexer-rzg6f
Name:         nrdee-indexer-rzg6f
Namespace:    asias-etl
Labels:       workflows.argoproj.io/completed=true
              workflows.argoproj.io/phase=Succeeded
Annotations:  <none>
API Version:  argoproj.io/v1alpha1
Kind:         Workflow
Metadata:
  Creation Timestamp:  2020-04-29T11:37:44Z
  Generate Name:       nrdee-indexer-
  Generation:          1
  Resource Version:    180809477
  Self Link:           /apis/argoproj.io/v1alpha1/namespaces/asias-etl/workflows/nrdee-indexer-rzg6f
  UID:                 d779f4ab-8a0d-11ea-b235-00505688fd38
Spec:
  Arguments:
  Entrypoint:            run-nrdee-indexer
  Service Account Name:  argo
  Templates:
    Arguments:
    Inputs:
    Metadata:
    Name:  run-nrdee-indexer
    Outputs:
    Steps:
      [map[arguments:map[] name:nrdee-indexer template:run-indexer-process]]
    Arguments:
    Container:
      Image:  docker-registry-default.okd.mitre.org/asias-etl/nrdee-indexer:latest
      Name:   nrdee-indexer
      Resources:
        Limits:
          Cpu:     1
          Memory:  5Gi
        Requests:
          Cpu:     1
          Memory:  5Gi
      Volume Mounts:
        Mount Path:  /etc/confd/.container/secrets
        Name:        secret
        Mount Path:  /var/lib/nrdee/solr
        Name:        asias-etl-nrdee-solr-volume-pv
    Inputs:
    Metadata:
    Name:  run-indexer-process
    Outputs:
  Volumes:
    Name:  secret
    Secret:
      Secret Name:  app-confd-secrets-nrdee-indexer
    Name:           asias-etl-nrdee-solr-volume-pv
    Persistent Volume Claim:
      Claim Name:  asias-etl-nrdee-solr-volume
Status:
  Conditions:
    Status:     True
    Type:       Completed
  Finished At:  2020-04-29T11:58:21Z
  Nodes:
    Nrdee - Indexer - Rzg 6 F:
      Children:
        nrdee-indexer-rzg6f-2973924931
      Display Name:  nrdee-indexer-rzg6f
      Finished At:   2020-04-29T11:58:21Z
      Id:            nrdee-indexer-rzg6f
      Name:          nrdee-indexer-rzg6f
      Outbound Nodes:
        nrdee-indexer-rzg6f-1673954213
      Phase:          Succeeded
      Started At:     2020-04-29T11:37:44Z
      Template Name:  run-nrdee-indexer
      Type:           Steps
    Nrdee - Indexer - Rzg 6 F - 1673954213:
      Boundary ID:    nrdee-indexer-rzg6f
      Display Name:   nrdee-indexer
      Finished At:    2020-04-29T11:58:20Z
      Id:             nrdee-indexer-rzg6f-1673954213
      Name:           nrdee-indexer-rzg6f[0].nrdee-indexer
      Phase:          Succeeded
      Started At:     2020-04-29T11:37:44Z
      Template Name:  run-indexer-process
      Type:           Pod
    Nrdee - Indexer - Rzg 6 F - 2973924931:
      Boundary ID:  nrdee-indexer-rzg6f
      Children:
        nrdee-indexer-rzg6f-1673954213
      Display Name:   [0]
      Finished At:    2020-04-29T11:58:21Z
      Id:             nrdee-indexer-rzg6f-2973924931
      Name:           nrdee-indexer-rzg6f[0]
      Phase:          Succeeded
      Started At:     2020-04-29T11:37:44Z
      Template Name:  run-nrdee-indexer
      Type:           StepGroup
  Phase:              Succeeded
  Started At:         2020-04-29T11:37:44Z
Events:
  Type    Reason             Age   From                 Message
  ----    ------             ----  ----                 -------
  Normal  WorkflowRunning    27m   nrdee-indexer-rzg6f  Workflow Running
  Normal  WorkflowSucceeded  6m    nrdee-indexer-rzg6f  Workflow completed
```

---

**Message from the maintainers**:

If you are impacted by this bug please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by jackiemazy** [comment]
Things I can think of for now are
* Too much logs in the pod main container
* Didn't see the `deadline monitor stopped` message in the wait container logs

**Comment by alexec** [comment]
This smells:

```
W0429 12:00:08.147863       1 reflector.go:302] pkg/mod/k8s.io/client-go@v0.0.0-20191225075139-73fd2ddc9180/tools/cache/reflector.go:98: watch of *unstructured.Unstructured ended with: The resourceVersion for the provided watch is too old.
```

**Comment by jackiemazy** [comment]
@alexec Thanks for looking into this... It does look like something odd. And I have seen those logs from the workflow-controller since I upgraded to argo 2.7.4. But other workflows seem to work fine (I have some multiple-step workflows went thru even with those messages in the controller). So it doesn't seem to be causing the workflow to hang.

**Comment by alexec** [comment]
https://sharingtechknowledge.blogspot.com/2019/03/openshift-resourceversion-for-provided.html

**Comment by jackiemazy** [comment]
Good to know that message isn't a bug. Now back to this issue, the workflow seems to stuck with no reasons, so am i :-(.

**Comment by logicfox** [comment]
I recently upgraded to 2.7 from 2.4 and seeing similar behaviour. In my case, it seems to hang when it tries to update labels. 
```
time="2020-04-30T10:39:08Z" level=info msg="Annotating pod with output"
```
In the above instance, that pod has been stuck there for over 10 minutes now. Sometimes, it stays stuck there for hours. There are other times where I see the same behaviour as OP - sidecars killed, annotation monitors stopped, etc, but the wait container still keeps running.

**Comment by jackiemazy** [comment]
Updating to v2.8.0-rc3 seems to get this problem solved.

**Comment by aviam** [comment]
This issue is back on version 3+

**Comment by zeppelinen** [comment]
Confirmed on 3.5.2
```
argo: v3.5.2+5b6ad2b.dirty
  BuildDate: 2023-11-27T18:22:00Z
  GitCommit: 5b6ad2be163ecd3f0251a931ab84dba3c6085ad2
  GitTreeState: dirty
  GitTag: v3.5.2
  GoVersion: go1.21.4
  Compiler: gc
  Platform: darwin/arm64
```

**Comment by Joibel** [comment]
Adding comments to closed issues won't get them looked at. Please open a new issue.
